src/components/interview.ts

```diff
@@ -36,41 +52,72 @@ export async function handleInterview(message: Discord.Message, args: string[],
 }
 
 async function help(message: Discord.Message): Promise<void> {
+  //should be deprecated as soon as proper help handler is built.
   const response =
     "I can't seem to recognize your command; the commands I know for interviewers are: \n" +
-    ' ```.interviewer signup [calendly/xai link] ``` \n' +
-    ' ```.interviewer list ```';
+    ' ```.interviewer signup [calendly/xai link] \n//signs up or update your calendar link```' +
+    ' ```.interviewer list (domain) \n//shows list of available interviewers, with optional domain filter```' +
+    ' ```.interviewer domain [domain] \n//adds or removes you from a domain``` ';
   await message.channel.send(response);
 }
 
 async function addInterviewer(message: Discord.Message, args: string[]): Promise<void> {
   const db = await openDB();
   const id = message.author.id;
+
+  //checks for existence of link arg
   const link = args.shift();
   if (!link) {
     help(message);
     return;
   }
+
+  //parses link and checks for validity
   const parsedLink = parseLink(link);
   if (!parsedLink) {
     message.channel.send(`Hmmm... I don't seem to recognize your meeting link. Be sure to use calendly or x.ai.`);
     return;
   }
+
+  //checks if user is already an interviewer, adds/updates info accordingly
   const res = await db.get('SELECT * FROM interviewers WHERE user_id = ?', id);
   if (!res) {
     db.run('INSERT INTO interviewers (user_id, link) VALUES(? , ?)', id, parsedLink);
-    message.channel.send(`<@${id}>, your info has been added. Thanks for signing up to help out! :codeyLove:`);
+    message.channel.send(`<@${id}>, your info has been added. Thanks for helping out! :codeyLove:`);
   } else {
     db.run('UPDATE interviewers SET link = ? WHERE user_id = ?', parsedLink, id);
     message.channel.send(`<@${id}>, your info has been changed.`);
   }
 }
 
-async function listInterviewers(message: Discord.Message, client: Discord.Client): Promise<void> {
+async function listInterviewers(message: Discord.Message, client: Discord.Client, args: string[]): Promise<void> {
   const db = await openDB();
-  const res = _.shuffle(await db.all('SELECT * FROM interviewers')) as Interviewer[];
+  let res: Interviewer[];
+  let domain = args.shift();
```

Comment:
same here, should we maybe validate the number of arguments supplied? if all these commands only require 1 or less arguments, maybe we could do a top level check somewhere for all of these subcommands

Reply:
maybe like `const domain = args.length == 1 ? args[0].toLowerCase() else null` if you are planning to check number of args here